pkg/apis/config/loader/loader.go

```diff
@@ -15,33 +15,25 @@
 package loader
 
 import (
+	"fmt"
 	"io/ioutil"
 
 	"github.com/gardener/gardener-extension-provider-aws/pkg/apis/config"
 	"github.com/gardener/gardener-extension-provider-aws/pkg/apis/config/install"
 
 	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/apimachinery/pkg/runtime/schema"
-	"k8s.io/apimachinery/pkg/runtime/serializer/json"
-	"k8s.io/apimachinery/pkg/runtime/serializer/versioning"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
 )
 
 var (
-	Codec  runtime.Codec
 	Scheme *runtime.Scheme
+	Codecs serializer.CodecFactory
 )
 
 func init() {
 	Scheme = runtime.NewScheme()
 	install.Install(Scheme)
-	yamlSerializer := json.NewYAMLSerializer(json.DefaultMetaFactory, Scheme, Scheme)
-	Codec = versioning.NewDefaultingCodecForScheme(
-		Scheme,
-		yamlSerializer,
-		yamlSerializer,
-		schema.GroupVersion{Version: "v1alpha1"},
-		runtime.InternalGroupVersioner,
-	)
+	Codecs = serializer.NewCodecFactory(Scheme, serializer.EnableStrict)
```

Comment:
Any special reason we need the strict mode here? I've never done this in any Gardener code before. Did we miss something?